plugins/icepak_utils.py

```
@@ -0,0 +1,225 @@
+        "ZVelocity": zvelocity_list,
+        "Energy": energy_list,
+        "DOIntensity": dointensity_list,
+        "JouleHeating": jouleheating_list,
+    })
+
+    # Drop columns with all NaN values (e.g., Temperature if not present in this file)
+    df = df.dropna(axis=1, how="all")
+
+    return df
+
+def make_combined_df(dict_monfiles):
+    """
+    Combines the monitor data from different .sd files into a single DataFrame based on 'Iteration'.
+    
+    Args:
+        dict_monfiles (dict): Dictionary of monitor names and corresponding .sd file paths.
+
+    Returns:
+        pd.DataFrame: A DataFrame with the combined monitor data.
+    """
+    
+    df_monfiles_total = None
+    
+    # Loop over each monitor file path in the dictionary
+    for monitor_name, file_path in dict_monfiles.items():
+        if file_path:
+            try:
+                # Parse the MON file into a DataFrame
+                df = parse_monfile_precise(file_path)
+                
+                # If it's the "Residual" monitor, merge without changing column names
+                if monitor_name == "Residual":
+                    df_residual = df.copy()
+                    
+                    if df_monfiles_total is None:
+                        df_monfiles_total = df_residual
+                    else:
+                        df_monfiles_total = pd.merge(df_monfiles_total, df_residual, on='Iteration', how='outer')
+                else:
+                    # Rename the second column with the monitor name to avoid duplicate column names
+                    df_non_residual = df.copy()
+                    df_non_residual.rename(columns={df.columns[1]: f"{df.columns[1]}-{monitor_name}"}, inplace=True)
+                    
+                    # Merge with the existing DataFrame
+                    df_monfiles_total = pd.merge(df_monfiles_total, df_non_residual, on='Iteration', how='outer')
+                    
+            except Exception as e:
```

Do not swallow the exception here. Remove the try/except - we will handle those in the `get_df()` function.